src/aws-cpp-sdk-core/include/smithy/identity/resolver/built-in/DefaultAwsCredentialIdentityResolver.h

```diff
@@ -9,33 +9,51 @@
 #include <aws/core/auth/AWSCredentials.h>
 #include <aws/core/auth/AWSCredentialsProviderChain.h>
 
+namespace{
+    constexpr char ALLOC_ID[] = "DefaultAwsCredentialIdentityResolver";
+
+}
+
 namespace smithy {
     /**
      * A smithy SigV4 AWS Credentials resolver wrapper on top of legacy SDK Credentials provider
      * TODO: refactor into own signer using smithy design
      */
     class DefaultAwsCredentialIdentityResolver : public AwsCredentialIdentityResolver {
+    protected:
+
+        mutable std::shared_ptr<Aws::Auth::AWSCredentialsProviderChain> legacyChain_sp;
+
+        //mutable Aws::Auth::DefaultAWSCredentialsProviderChain legacyCredentialsProvider{};
```

Comment:
remove commented code, also why did you move this member to the top of the class defintion?